compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt

```diff
@@ -251,24 +297,46 @@ private class BackendCodeGenerator(
 
             is UpdateNode -> codeGenerator.update(protocol, stmt)
 
-            is OutParameterInitializationNode -> outParameterInitialization()
+            is OutParameterInitializationNode -> outParameterInitialization(protocol, stmt)
 
             is OutputNode -> CodeBlock.of(
                 "runtime.output(%T(%L))",
                 typeAnalysis.type(stmt.message).valueClass,
                 codeGenerator.exp(protocol, stmt.message)
             )
         }
-    }
 
-    fun outParameterInitialization(
-        /* protocol: Protocol, stmt: OutParameterInitializationNode */
-    ): CodeBlock =
-        CodeBlock.of("") // TODO (merge from fn calls)
+    private fun outParameterInitialization(
+        protocol: Protocol,
+        stmt: OutParameterInitializationNode
+    ): CodeBlock {
+        val rhs = when (val init = stmt.initializer) {
+            is OutParameterConstructorInitializerNode -> codeGenerator.constructorCall(
+                protocol,
+                init.objectType,
+                init.arguments
+            )
+            is OutParameterExpressionInitializerNode -> codeGenerator.exp(protocol, init.expression)
+        }
+        val parameterName = context.kotlinName(stmt.name.value)
+
+        return CodeBlock.of( // TODO this ends up looking strange
```

Comment:
To fix, return a list of code blocks and add them both to the function list.

Or, since it would be pain to change SimpleStatement function above, just do semicolon.